src/platform/graphics/webgl/webgl-graphics-device.js

```diff
@@ -2062,6 +2062,63 @@ class WebglGraphicsDevice extends GraphicsDevice {
         gl.readPixels(x, y, w, h, gl.RGBA, gl.UNSIGNED_BYTE, pixels);
     }
 
+    /**
+     * Asynchronously reads a block of pixels from a specified rectangle of the current color framebuffer
+     * into an ArrayBufferView object.
+     *
+     * @param {number} x - The x-coordinate of the rectangle's lower-left corner.
+     * @param {number} y - The y-coordinate of the rectangle's lower-left corner.
+     * @param {number} w - The width of the rectangle, in pixels.
+     * @param {number} h - The height of the rectangle, in pixels.
+     * @param {ArrayBufferView} pixels - The ArrayBufferView object that holds the returned pixel
+     * data.
+     * @ignore
+     */
+    async readPixelsAsync(x, y, w, h, pixels) {
+        const gl = this.gl;
+
+        const clientWaitAsync = (flags, interval_ms) => {
+            const sync = gl.fenceSync(gl.SYNC_GPU_COMMANDS_COMPLETE, 0);
```

Comment:
this is WebGL2 only, do we need some GL1 fallback?

Reply:
yeah good point. we should probably just return synchronously